script/validate/vendor

```diff
@@ -2,19 +2,8 @@
 
 set -eu -o pipefail
 
-rm -rf vendor/
-vndr |& grep -v -i clone
-
-DIFF_PATH="vendor/"
-DIFF=$(git status --porcelain -- "$DIFF_PATH")
-
-if [ "$DIFF" ]; then
-    echo
-    echo "These files were modified:"
-    echo
-    echo "$DIFF"
-    echo
-    exit 1
-else
-    echo "$DIFF_PATH is correct"
-fi
+echo "- Checking for any unused/missing packages in go.mod..."
+GO111MODULE=on go mod tidy
+echo "- Checking for unused packages in vendor..."
+GO111MODULE=on go mod vendor
+git diff --exit-code -- go.sum go.mod vendor/
```

Comment:
^ looks like you missed a newline at the end

Reply:
Thanks!

Reply:
Nice. Looking good @tariq1890. One last case that we need to check. `git diff` will only care about tracked files which means that if someone forgets to add a vendored directory this will still exit cleanly. You should be able to check with something like

```
untracked=$(git ls-files --others vendor | wc -l | awk '{ print $1 }')
[ "$untracked" == "0" ]
```